// letterboxd-search/letterboxd-search.js
const request = require('request-promise');
const cheerio = require('cheerio');
const numeral = require('numeral');

module.exports = { search };

function search(title) {
  return new Promise(function(resolve, reject) {
    const endpoint = title.toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g," ").replace(/ /g,'-')
    debugger;
    request({
      method: "GET",
      url: `https://letterboxd.com/film/${endpoint}`,
      transform(body) {
        const ret = cheerio.load(body)
        return ret;
      }
    }).then($ => {

      let temp = $('.js-video-zoom').attr('href');
      let trailer = temp ? "https://youtu.be/" + $('.js-video-zoom').attr('href').split('/')[4].split('?')[0] : "Unavailable"

      let obj = {
        title: $('meta[name="twitter:title"]').attr('content'),
        url: $('meta[property="og:url"]').attr('content'),
        year: numeral($('.film-poster').attr('data-film-release-year')).value(),
        director: $('meta[name="twitter:data1"]').attr('content'),
        runtimeMinutes: numeral($('.text-link').text().trim()).value(),
        synopsis: $('meta[property="og:description"]').attr('content'),
        averageRating: String($('meta[name="twitter:data2"]').attr('content')).slice(0,4),
        trailer: trailer,
        backdropImage: $('meta[property="og:image"]').attr('content'),
        posterImage: $('.film-poster img').attr('src'),
        statusCode: 200
      }
      resolve(obj);
    }).catch(err => {
      console.log(err);
      reject({
        name: err.name,
        statusCode: err.statusCode,
      });
    });
  });
}
